applications/matter_bridge/src/bridge_shell.cpp

```diff
@@ -20,8 +21,19 @@ static int add_bridged_device(const struct shell *shell, size_t argc, char **arg
 		nodeLabel = argv[2];
 	}
 
-	CHIP_ERROR err =
-		GetBridgeManager().AddBridgedDevice(static_cast<BridgedDevice::DeviceType>(deviceType), nodeLabel);
+	auto *newBridgedDevice = BridgeFactory::GetBridgedDeviceFactory().Create(
+		static_cast<BridgedDevice::DeviceType>(deviceType), nodeLabel);
+
+	VerifyOrReturnError(newBridgedDevice != nullptr, -EPERM,
+			    shell_fprintf(shell, SHELL_INFO, "Cannot allocate Matter device of given type\n"));
+
+	auto *newDataProvider = BridgeFactory::GetDataProviderFactory().Create(
+		static_cast<BridgedDevice::DeviceType>(deviceType), BridgeManager::HandleUpdate);
+
+	VerifyOrReturnError(newDataProvider != nullptr, -EPERM,
```

Comment:
My point was that `newBridgedDevice` should be released in the case of error here.

Reply:
Fixed.